### ISSUE TITLE
How to Select a single item?

### ISSUE THREAD

**Issue by kennycason** [JetBrains/Exposed#549]
I found a few examples online showing the usage of `findById` but I can not find any such function (or anything starting with find) when used in conjunction with an IntIdTable. I saw the select function but can't even find a clean way to "get a single row" from the database. I'm not sure how to cleanly convert from ResultRow. 

I've greatly enjoyed using Exposed for batch loading lots of data but have found the api a bit unclear when getting started for a basic crud app, even with IntelliJ magic. Save/Update, no problem. selects/select by id, less clear. I think the readme could benefit from a few more examples of basic crud examples.

**Comment by kennycason** [comment]
I understand how to use ResultRow for a list of things since I can just iterate over, but am not sure how to cleanly get a single Row. I can't just do ` User.select { User.auth0UserId.eq(auth0UserId) }.first()` since that's not safe. I also don't see a "result count" option or any option to return a ResultRow? I'm just a bit lost on how to "fetch a single item"

**Comment by kennycason** [comment]
I guess firstOrNull() works. but still seems a bit verbose for single row lookup by id. So i'm assuming I'm either missing something, or exposed could benefit from single row lookup by id.

**Comment by Tapac** [comment]
First of all - did you read [wiki](https://github.com/JetBrains/Exposed/wiki) or just readme file? There are much more examples and cases covered.

Also, there is no specific function to fetch only a single row from Query, but you can define it in your code and use where it's needed.
```kotlin
fun Query.firstRow() : ResultRow? = limit(1).firstOrNull()
```

**Comment by kennycason** [comment]
Thanks for the response. I did read the wiki and initially did not find what I was looking for. I also realized I actually had two problems. I was looking for something unreasonable in that result will always be a ResultRow and not a domain object. 

e.g. 
```
object Email : Table("email") {
    val id = Email.integer("id").autoIncrement().primaryKey()
    val email = Email.varchar("email", 256)
    val createdAt = Email.datetime("created_at")
}
```
I was also wanting to just get an Email *instance* back, but obviously that doesn't make much sense given it's an `object`.

I was looking for something as simple as Hibernate/JPA:
```
interface EmailRepository : JpaRepository<Email, Long>
```
where i get `findById(): Email`. 

So two things: 
- I was wanting to easily get a single row (can do so view defining my own function like you mentioned)
- I was also wanting to return domain objects, which I realized after more reading I'd have to construct them manually from the ResultRow for all reads. Which I think is likely faster than all the reflection stuff Hibernate is doing under the hood, but obviously a bit more tedious.

I'll keep poking around. :)
